src/overlay021/pokedex_text.c

```diff
@@ -0,0 +1,179 @@
+#include "overlay021/pokedex_text.h"
+
+#include <nitro.h>
+#include <string.h>
+
+#include "constants/narc.h"
+#include "constants/species.h"
+
+#include "text/pl_msg.naix"
+
+#include "error_handling.h"
+#include "inlines.h"
+#include "message.h"
+#include "message_util.h"
+#include "pokedex_language.h"
+#include "strbuf.h"
+
+enum LangIndex {
+    LI_JAPANESE,
+    LI_ENGLISH,
+    LI_FRENCH,
+    LI_GERMAN,
+    LI_ITALIAN,
+    LI_SPANISH
+};
+
+static const u8 messageBankLanguageOrder[NUM_LANGUAGES] = {
+    LI_ENGLISH,
+    LI_FRENCH,
+    LI_GERMAN,
+    LI_ITALIAN,
+    LI_SPANISH,
+    LI_JAPANESE
+};
+
+static inline int LanguageIndex_Guarded(int languageIndex);
+static inline BOOL ValidLanguage(int species, int languageIndex);
+static Strbuf *LoadMessage(int bankID, int entryID, int heapID);
+static void GetLanguageIndex(int species, int language, int *dexNum, int *languageIndexUnguarded, int *languageIndex);
+
+void PokedexText_Free(Strbuf *strbuf)
+{
+    Strbuf_Free(strbuf);
+}
+
+int PokedexText_ForeignLanguage(int languageIndex)
+{
+    return PokedexLanguage_IndexToLanguage(messageBankLanguageOrder[languageIndex + 1]);
+}
+
+Strbuf *PokedexText_NameNumber(int species, int language, int heapID)
+{
+    int dexNum;
+    int languageIndex_unguarded;
+    int index;
+    int bankID;
+
+    GetLanguageIndex(species, language, &dexNum, &languageIndex_unguarded, &index);
+
+    if (index == NUM_LANGUAGES) {
+        return MessageUtil_SpeciesName(species, heapID);
+    } else {
+        const int nameNumberBanks[] = {
+            message_bank_unk_0717,
+            message_bank_species_name_number_1,
+            message_bank_unk_0713,
+            message_bank_unk_0714,
+            message_bank_species_name_number_2,
+            message_bank_species_name_number_3
```

Comment:
**question**: Can we name these banks differently now? Applies throughout the file.

Reply:
Ah, is this handled by #324?

Reply:
Yes. I was planning to do piecemeal PRs and I though it would be slightly easier to rename in the PR focused on those files specifically.